packages/dos-donts-block/src/components/ImageComponent.tsx

```diff
@@ -37,30 +37,35 @@ const ImageComponent = ({
             ? toRgbaString(backgroundColor)
             : '';
 
-    const getImage = () => {
-        const width = (isEditing ? 800 : 800 / columns) * (window?.devicePixelRatio ?? 1);
-
-        return src ? (
-            <div className="tw-relative tw-w-full">
+    const getImage = () =>
+        image ? (
+            <div
+                {...draggableProps}
+                ref={setContainerRef}
+                className={joinClassNames([
+                    'tw-relative tw-w-full',
+                    isEditing && (isDragging ? 'tw-cursor-grabbing' : 'tw-cursor-grab'),
+                ])}
+            >
                 <div
                     style={{ height: imageHeight, borderRadius, border, background }}
                     className="tw-w-full tw-overflow-hidden"
                 >
-                    <img
-                        {...draggableProps}
-                        className={joinClassNames([
-                            'tw-w-full tw-h-full',
-                            isEditing && (isDragging ? 'tw-cursor-grabbing' : 'tw-cursor-grab'),
-                            (imageHeightChoice !== DoDontImageHeight.Auto || isCustomImageHeight) &&
-                                imageDisplay === ImageFitChoice.FILL &&
-                                'tw-object-cover',
-                            (imageHeightChoice !== DoDontImageHeight.Auto || isCustomImageHeight) &&
-                                imageDisplay === ImageFitChoice.FIT &&
-                                'tw-object-contain',
-                        ])}
-                        src={src.replace('{width}', width.toString())}
-                        alt=""
-                    />
+                    {containerWidth && (
+                        <ResponsiveImage
+                            containerWidth={containerWidth}
+                            image={image}
+                            className={joinClassNames([
+                                'tw-h-full',
```

Comment:
We could introduce this object-fit prop but I think it might not add too much, so for now I'd prefer to keep it simple with just the className and style tag being exposed.